src/python/pants/option/global_options.py

```diff
@@ -90,6 +90,7 @@ class ExecutionOptions:
     remote_store_initial_timeout: int
     remote_store_timeout_multiplier: float
     remote_store_maximum_timeout: int
+    remote_store_eager_fetch: bool
```

Comment:
I'd recommend namespacing this as `remote_cache_eager_fetch`, as that's the only codepath it will likely (ever) impact.